src/v/cloud_storage/topic_manifest.cc

```diff
@@ -1299,4 +1301,15 @@ remote_manifest_path topic_manifest::get_manifest_path() const {
     return get_topic_manifest_path(
       _topic_config->tp_ns.ns, _topic_config->tp_ns.tp);
 }
+
+bool topic_manifest::is_mapping_updated() const noexcept {
```

Comment:
how about `is_mapping_complete`?

Reply:
> the check is perfomed in a unit test to not block compilation

why not block it? having an incomplete mapping is a bug, why shouldn't we prevent compilation?

Reply:
`how about is_mapping_complete?`
it is complete now, but maybe we will want to leave something out in the future. I want to communicate that "the serialization code is updated to the latest version of topic_properties. Whether it's complete or not, it's an implementation detail". naming is hard.

`why not block it?`
I imagine the case of somebody adding a topic property, and it will not be something significantly complex. So adding it when the test is failing should be easy and a better devex than having compilation fail

Reply:
They can comment out the assertion if they want a yolo build. I'd argue that this is a better devex. I often change code and run ducktape directly. If I'm making a change here and I don't run unit tests then I'd spend time debugging what went wrong instead of compiler telling me straight ahead that i missed something.